internal/controllers/import_controller_v3.go

```diff
@@ -0,0 +1,393 @@
+	log := log.FromContext(ctx)
+
+	// placeholder name/namespace must be set to be able to test this but it won't be used.
+	rancherCluster := &managementv3.Cluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "c-",
+			Namespace: capiCluster.Namespace,
+			Labels: map[string]string{
+				capiClusterOwner:          capiCluster.Name,
+				capiClusterOwnerNamespace: capiCluster.Namespace,
+			},
+		},
+	}
+
+	rancherClusterList := &managementv3.ClusterList{}
+	selectors := []client.ListOption{
+		client.MatchingLabels{
+			capiClusterOwner:          capiCluster.Name,
+			capiClusterOwnerNamespace: capiCluster.Namespace,
+			ownedLabelName:            "",
+		},
+	}
+	err := r.RancherClient.List(ctx, rancherClusterList, selectors...)
+
+	if client.IgnoreNotFound(err) != nil {
+		log.Error(err, fmt.Sprintf("Unable to fetch rancher cluster %s", client.ObjectKeyFromObject(rancherCluster)))
+		return ctrl.Result{Requeue: true}, err
+	}
+
+	if len(rancherClusterList.Items) != 0 {
+		if len(rancherClusterList.Items) > 1 {
+			log.Info("More than one rancher cluster found. Will default to using the first one.")
+		}
+
+		rancherCluster = &rancherClusterList.Items[0]
+	}
+
+	if !capiCluster.ObjectMeta.DeletionTimestamp.IsZero() {
+		if err := r.deleteDependentRancherCluster(ctx, capiCluster); err != nil {
+			return ctrl.Result{Requeue: true}, fmt.Errorf("error deleting associated managementv3.Cluster resources: %w", err)
+		}
+	}
+
+	if !rancherCluster.ObjectMeta.DeletionTimestamp.IsZero() {
+		return r.reconcileDelete(ctx, capiCluster)
+	}
+
+	return r.reconcileNormal(ctx, capiCluster, rancherCluster)
+}
+
+func (r *CAPIImportManagementV3Reconciler) reconcileNormal(ctx context.Context, capiCluster *clusterv1.Cluster,
+	rancherCluster *managementv3.Cluster,
+) (ctrl.Result, error) {
+	log := log.FromContext(ctx)
+
+	err := r.RancherClient.Get(ctx, client.ObjectKeyFromObject(rancherCluster), rancherCluster)
+	if apierrors.IsNotFound(err) {
+		shouldImport, err := util.ShouldAutoImport(ctx, log, r.Client, capiCluster, importLabelName)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+
+		if !shouldImport {
+			log.Info("not auto importing cluster as namespace or cluster isn't marked auto import")
+			return ctrl.Result{}, nil
+		}
+
+		if err := r.RancherClient.Create(ctx, &managementv3.Cluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace:    capiCluster.Namespace,
+				GenerateName: "c-",
+				Labels: map[string]string{
+					capiClusterOwner:          capiCluster.Name,
+					capiClusterOwnerNamespace: capiCluster.Namespace,
+					ownedLabelName:            "",
+				},
+			},
+			Spec: managementv3.ClusterSpec{
+				DisplayName: capiCluster.Name,
+				Description: "CAPI cluster imported to Rancher",
+			},
+		}); err != nil {
+			return ctrl.Result{}, fmt.Errorf("error creating rancher cluster: %w", err)
+		}
+
+		return ctrl.Result{Requeue: true}, nil
```

Comment:
```suggestion
rancherCluster = createCopy
```
Don’t think we need to return here. Rancher cluster was created and stored, we just need to swap the pointers with the copy.